superset-frontend/src/dashboard/components/menu/ShareMenuItems/index.tsx

```diff
@@ -57,48 +53,29 @@ const ShareMenuItems = (props: ShareMenuItemProps) => {
     ...rest
   } = props;
 
-  const tabId = useTabId();
-
-  const getShortUrl = useUrlShortener(url || '');
-
-  async function getCopyUrl() {
-    const risonObj = getUrlParam(URL_PARAMS.nativeFilters);
-    if (typeof risonObj === 'object' || !dashboardId) return null;
-    const prevData = await getFilterValue(
-      dashboardId,
-      getUrlParam(URL_PARAMS.nativeFiltersKey),
-    );
-    const newDataMaskKey = await createFilterKey(
-      dashboardId,
-      JSON.stringify(prevData),
-      tabId,
-    );
-    const newUrl = new URL(`${window.location.origin}${url}`);
-    newUrl.searchParams.set(URL_PARAMS.nativeFilters.name, newDataMaskKey);
-    return `${newUrl.pathname}${newUrl.search}`;
-  }
-
   async function generateUrl() {
+    // chart
     if (formData) {
-      const key = await postFormData(
-        parseInt(formData.datasource.split('_')[0], 10),
-        formData,
-        formData.slice_id,
-        tabId,
-      );
-      return `${window.location.origin}${mountExploreUrl(null, {
-        [URL_PARAMS.formDataKey.name]: key,
-        [URL_PARAMS.sliceId.name]: formData.slice_id,
-      })}`;
+      return getChartPermalink(formData.slice_id, formData);
+    }
+    // dashboard
+    const nativeFiltersKey = getUrlParam(URL_PARAMS.nativeFiltersKey);
+    let filterState = {};
+    if (nativeFiltersKey && dashboardId) {
+      filterState = await getFilterValue(dashboardId, nativeFiltersKey);
     }
-    const copyUrl = await getCopyUrl();
-    return getShortUrl(copyUrl);
+    return getDashboardPermalink(String(dashboardId), filterState);
   }
 
   async function onCopyLink() {
     try {
-      await copyTextToClipboard(await generateUrl());
-      addSuccessToast(t('Copied to clipboard!'));
+      const url = await generateUrl();
+      if (url) {
+        await copyTextToClipboard(url);
+        addSuccessToast(t('Copied to clipboard!'));
+      } else {
+        addDangerToast(t('Dashboard missing'));
```

Comment:
The lack of an URL can only be caused by a missing dashboard? If not, maybe make the message more generic?

Reply:
Actually this can't return a falsy value anymore, so removing this.